pyuvdata/uvcal/uvcal.py

```diff
@@ -615,8 +551,136 @@ def __init__(self):
             "filename", required=False, description=desc, expected_type=str
         )
 
+        # initialize the telescope object
+        self.telescope = Telescope()
+
+        # set the appropriate telescope attributes as required
+        self.telescope._set_uvcal_requirements()
+
         super(UVCal, self).__init__()
 
+    @property
+    def telescope_name(self):
+        """The telescope name (stored on the Telescope object internally)."""
+        return self.telescope.name
+
+    @telescope_name.setter
+    def telescope_name(self, val):
+        self.telescope.name = val
+
+    @property
+    def instrument(self):
+        """The instrument name (stored on the Telescope object internally)."""
+        return self.telescope.instrument
+
+    @instrument.setter
+    def instrument(self, val):
+        self.telescope.instrument = val
+
+    @property
+    def telescope_location(self):
+        """The telescope location (stored on the Telescope object internally)."""
+        return self.telescope.location
```

Comment:
Hmmm, since there are a few of these, maybe it's easier to move them to the `__getattr__` and `__setattr__` methods, since that'll be a couple of lines of code. I don't love doing that, but I think in this case it might be worth it. In the spirit of "there should be one right way of doing it" I feel like we should also discourage using these old-style names (`telescope_name` etc) in favor of `telescope.name`. We could consider adding a deprecation warning to the `__getattr__`  method if we like (or, in the future?). My other gripe with `__getattr__` is that it doesn't play well with code editors auto-completion (including jupyter). But again, in this case, I don't think we want to encourage using these attribute names anyhow.

Reply:
Yeah, I wasn't loving how many there are and how they're replicated across the objects. I'm was also thinking about whether I could move them to UVBase, but not all objects that inherit from UVBase have a telescope object.

I guess we hadn't decided that we wanted to move people away from the older names, that's a very big adjustment. We were intending this change to largely be behind the scenes and to affect the users as little as possible.

Reply:
These are now in `_getattr__` and `__setattr__` methods on UVBase. I haven't deprecated them yet.

Reply:
they are now deprecated.